lib/sys/stdint.d

```diff
@@ -0,0 +1,131 @@
+
+    The aliases are:
+
+    $(ATABLE $(TR
+    $(TH Exact Alias)
+    $(TH Description)
+    $(TH At Least Alias)
+    $(TH Description)
+    $(TH Fast Alias)
+    $(TH Description)
+    )$(TR
+    $(TD int8_t)
+    $(TD exactly 8 bits signed)
+    $(TD int_least8_t)
+    $(TD at least 8 bits signed)
+    $(TD int_fast8_t)
+    $(TD fast 8 bits signed)
+    )$(TR
+    $(TD uint8_t)
+    $(TD exactly 8 bits unsigned)
+    $(TD uint_least8_t)
+    $(TD at least 8 bits unsigned)
+    $(TD uint_fast8_t)
+    $(TD fast 8 bits unsigned)
+
+    )$(TR
+    $(TD int16_t)
+    $(TD exactly 16 bits signed)
+    $(TD int_least16_t)
+    $(TD at least 16 bits signed)
+    $(TD int_fast16_t)
+    $(TD fast 16 bits signed)
+    )$(TR
+    $(TD uint16_t)
+    $(TD exactly 16 bits unsigned)
+    $(TD uint_least16_t)
+    $(TD at least 16 bits unsigned)
+    $(TD uint_fast16_t)
+    $(TD fast 16 bits unsigned)
+
+    )$(TR
+    $(TD int32_t)
+    $(TD exactly 32 bits signed)
+    $(TD int_least32_t)
+    $(TD at least 32 bits signed)
+    $(TD int_fast32_t)
+    $(TD fast 32 bits signed)
+    )$(TR
+    $(TD uint32_t)
+    $(TD exactly 32 bits unsigned)
+    $(TD uint_least32_t)
+    $(TD at least 32 bits unsigned)
+    $(TD uint_fast32_t)
+    $(TD fast 32 bits unsigned)
+
+    )$(TR
+    $(TD int64_t)
+    $(TD exactly 64 bits signed)
+    $(TD int_least64_t)
+    $(TD at least 64 bits signed)
+    $(TD int_fast64_t)
+    $(TD fast 64 bits signed)
+    )$(TR
+    $(TD uint64_t)
+    $(TD exactly 64 bits unsigned)
+    $(TD uint_least64_t)
+    $(TD at least 64 bits unsigned)
+    $(TD uint_fast64_t)
+    $(TD fast 64 bits unsigned)
+    ))
+
+    The ptr aliases are integral types guaranteed to be large enough
+    to hold a pointer without losing bits:
+
+    $(ATABLE $(TR
+    $(TH Alias)
+    $(TH Description)
+    )$(TR
+    $(TD intptr_t)
+    $(TD signed integral type large enough to hold a pointer)
+    )$(TR
+    $(TD uintptr_t)
+    $(TD unsigned integral type large enough to hold a pointer)
+    ))
+
+    The max aliases are the largest integral types:
+
+    $(ATABLE $(TR
+    $(TH Alias)
+    $(TH Description)
+    )$(TR
+    $(TD intmax_t)
+    $(TD the largest signed integral type)
+    )$(TR
+    $(TD uintmax_t)
+    $(TD the largest unsigned integral type)
+    ))
+
+ * Macros:
+ *  ATABLE=<table border="1" cellspacing="0" cellpadding="5">$0</table>
+ *
+ * Copyright: Copyright The D Language Foundation 2000 - 2009.
+ * License:   $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+ * Authors:   $(HTTP digitalmars.com, Walter Bright)
+ * Source:    $(PHOBOSSRC std/stdint.d)
+ */
+/*          Copyright The D Language Foundation 2000 - 2009.
+ * Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ */
+module lib.sys.stdint;
+
+public import core.stdc.stdint;
```

Comment:
We don't differentiate between the C standard library and Phobos?

Reply:
Apparently not. I just copied the file over from the existing Phobos file. Funny enough, every response to my announce post is about the same thing too. 😆

One possible side-effect of this process is that people finally get a good look at _all_ the crazy in Phobos that they've been not seeing for years.

Reply:
I would guess that Adam just grabbed some stuff from Phobos v2 that he thought would be simple and safe to grab. Personally, I didn't even realize that this was in there.

We previously made a concerted effort to remove stdc declarations from Phobos, and I guess that this got missed, because it wasn't under stdc. It should probably go as well. We have our own D types, and that's what Phobos should be using in its public APIs. There may be some cases where we decide that we do want to use some C types publicly in a D API (e.g. because it's giving low-level access to something), but that should be an exception rather than the rule.

Reply:
LOL. And Adam replied while I was writing up my reply.